core/ir/riscv64/decode.c

```diff
@@ -0,0 +1,247 @@
+/* **********************************************************
+ * Copyright (c) 2022 Rivos, Inc.  All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Rivos, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL RIVOS, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+#include "../globals.h"
+#include "instr.h"
+#include "decode.h"
+
+bool
+is_isa_mode_legal(dr_isa_mode_t mode)
+{
+    return (mode == DR_ISA_RV64IMAFDC);
+}
+
+app_pc
+canonicalize_pc_target(dcontext_t *dcontext, app_pc pc)
+{
+    return pc;
+}
+
+DR_API
+app_pc
+dr_app_pc_as_jump_target(dr_isa_mode_t isa_mode, app_pc pc)
+{
+    return pc;
+}
+
+DR_API
+app_pc
+dr_app_pc_as_load_target(dr_isa_mode_t isa_mode, app_pc pc)
+{
+    return pc;
+}
+
+byte *
+decode_eflags_usage(void *drcontext, byte *pc, uint *usage, dr_opnd_query_flags_t flags)
+{
+    ASSERT_NOT_IMPLEMENTED(false);
```

Comment:
Generally we put a TODO or XXX comment with the i#NNN for unfinished code.

Ditto below.

Reply:
Sure, I'll add it.
My reasoning for using only `ASSERT_NOT_IMPLEMENTED()` was that it is a RISC-V specific file so the issue comment seemed superfluous. Conversely in generic files I've put both the assert and the comment for the unimplemented RISC-V bits.

Reply:
I'm imagining that some of these will be implemented in this file but some will be left unimplemented b/c they're not exercised on code paths of some initial tests, and a long time later we have most of the file filled in but not all and someone reading it maybe can't tell whether something was deliberately decided to not be needed on RISC-V (though you'd expect a comment about that too) or it was just never looked at beyond the initial skeleton code.  It does add extra lines but IMHO it makes it seem more carefully considered.

Reply:
I've added the FIXME comment to all asserts.